diesel/src/query_builder/update_statement/changeset.rs

```diff
@@ -49,15 +40,37 @@ impl<T: AsChangeset> AsChangeset for Option<T> {
     }
 }
 
-impl<T: Changeset<DB>, DB: Backend> Changeset<DB> for Option<T> {
-    fn is_noop(&self) -> bool {
-        self.is_none()
-    }
+impl<Left, Right> AsChangeset for Eq<Left, Right>
+where
+    Left: Column,
+    Right: AppearsOnTable<Left::Table>,
+{
+    type Target = Left::Table;
+    type Changeset = Assign<Left, Right>;
 
-    fn walk_ast(&self, out: AstPass<DB>) -> QueryResult<()> {
-        match *self {
-            Some(ref c) => c.walk_ast(out),
-            None => Ok(()),
+    fn as_changeset(self) -> Self::Changeset {
+        Assign {
+            _column: self.left,
```

Comment:
Why does this use a underscore name?

Reply:
It's never actually used. Only the type is. I didn't feel like using `PhantomData` for it since we actually have the value at the point of construction.